test/index.ts

```diff
@@ -15,6 +16,30 @@ import {
   WritableKeys,
 } from "../lib";
 
+function testDeepPartial() {
+  type Params = {
+    [key: string]: any;
+  };
+
+  type Input = {
+    simple: number;
+    nested: {
+      date: Date;
+    };
+    params: Params;
+  };
+
+  type Expected = {
+    simple?: number;
+    nested?: {
+      date?: DeepPartial<Date>;
```

Comment:
Do not use `DeepPartial` in expected type since its still not clear what should be the result.
Probably the type should look like this:

```typescript
type Expected = {
    simple?: number;
    nested?: {
      date?: Date;
    };
    params?: {
      [key: string]?: any;
    };
  };
```

Reply:
`[key: string]?: any` is not valid, and I have not come up with a solution.

Reply:
Since `[key: string]` is already a partial property, I think we can directly use `[key: string]: any`